packages/mui-system/src/cssVars/cssVarsParser.ts

```diff
@@ -150,9 +148,10 @@ export default function cssVarsParser(
           assignNestedKeys(vars, keys, `var(${cssVar})`);
         }
       }
+      assignNestedKeys(parsedTheme, keys, value);
     },
     (keys) => keys[0] === 'vars', // skip 'vars/*' paths
   );
 
-  return { css, vars };
+  return { css, vars, parsedTheme };
```

Comment:
`parsedTheme` is the new object which will make functions inside baseTheme work.